javaslang/src/main/java/javaslang/Value.java

```diff
@@ -516,7 +516,7 @@ default CharSeq toCharSeq() {
      */
     @SuppressWarnings("unchecked")
     @GwtIncompatible("reflection is not supported")
-    default T[] toJavaArray(Class<T> componentType) {
+    default T[] toJavaArray(Class<? super T> componentType) {
```

Comment:
@jbgi It does also not help for generic classes :-/

```java
final Value<Option<Integer>> value = of(Option.of(1), Option.of(2), Option.of(3));

// Compile error: Cannot select from parameterized type
final Option<Integer>[] ints = value.toJavaArray(Option<Integer>[].class);
```

@paplorinc It seems that `Class<? super T>` is the only solution (but unsafe).

I'm tempted to take it. What do you think? +1 or -1 :)